From 8a51115b16ac550bf49094b3a136a8f18a65f99f Mon Sep 17 00:00:00 2001
From: Richard Gong <richard.gong@intel.com>
Date: Fri, 30 Apr 2021 08:45:34 -0500
Subject: [PATCH 29/41] HSD #22012990567-3: crypto: intel_fcs: clean up
 callback function

commit dbd15e5ad717ac95ef2d1b00d85474eccd192d7e from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.114-lts

Clean up fcs_vab_callback function.

Signed-off-by: Richard Gong <richard.gong@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/crypto/intel_fcs.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/crypto/intel_fcs.c b/drivers/crypto/intel_fcs.c
index f203113c13a5..e5a0ac06732e 100644
--- a/drivers/crypto/intel_fcs.c
+++ b/drivers/crypto/intel_fcs.c
@@ -82,17 +82,17 @@ static void fcs_vab_callback(struct stratix10_svc_client *client,
 {
 	struct intel_fcs_priv *priv = client->priv;
 
-	priv->status = 0;
-
-	if (data->status == BIT(SVC_STATUS_INVALID_PARAM)) {
-		priv->status = -EINVAL;
-		dev_warn(client->dev, "rejected, invalid param\n");
-	} else if (data->status == BIT(SVC_STATUS_ERROR)) {
+	if (data->status == BIT(SVC_STATUS_ERROR)) {
 		priv->status = *((unsigned int *)data->kaddr1);
 		dev_err(client->dev, "mbox_error=0x%x\n", priv->status);
 	} else if (data->status == BIT(SVC_STATUS_BUSY)) {
 		priv->status = -ETIMEDOUT;
 		dev_err(client->dev, "timeout to get completed status\n");
+	} else if (data->status == BIT(SVC_STATUS_OK)) {
+		priv->status = 0;
+	} else {
+		priv->status = -EINVAL;
+		dev_err(client->dev, "rejected, invalid param\n");
 	}
 
 	complete(&priv->completion);
-- 
2.18.1

